ddtrace/internal/agent.py

```diff
@@ -44,7 +45,19 @@ def get_trace_url():
 
     Raises a ``ValueError`` if the URL is not supported by the Agent.
     """
-    url = os.environ.get("DD_TRACE_AGENT_URL", "http://%s:%d" % (get_hostname(), get_trace_port()))
+    # check these envars inside of method to make testing easier
+    user_supplied_host = os.environ.get("DD_AGENT_HOST", os.environ.get("DATADOG_TRACE_AGENT_HOSTNAME")) is not None
+    user_supplied_port = os.environ.get("DD_AGENT_PORT", os.environ.get("DD_TRACE_AGENT_PORT")) is not None
+    url = os.environ.get("DD_TRACE_AGENT_URL")
+
+    if not url:
+        if user_supplied_host or user_supplied_port:
+            url = "http://%s:%s" % (get_hostname(), get_trace_port())
+        elif os.path.exists("/var/run/datadog/apm.socket"):
```

Comment:
Is it worth doing a bit more here to help in onboarding/configuring? I'm thinking that we could check the permissions (read/write, or whichever we need) and potentially warn the user.

Not quite sure the best course of action in the event that it's not read/writeable but we probably don't want to default to it if we know it won't work.

Reply:
I see what you're saying but I'm not entirely sure it's worth it. You'll get an error message anyways when the tracer tries to send traces to this address if there's some permissions issue, just like you would if the agent wasn't running and you tried to send traces over http. 

If you think we should still try to check this more, we can get the file permissions like so:

```
import os
import stat

st = os.stat(yourfile)
oct_perm = oct(st.st_mode)
print(oct_perm)
```
The last 3 digits of the output are the file permissions, however we don't know what permissions are required for the actual application because we don't know how it's being run, right? It could be under the user, the group, or all.
https://pymotw.com/2/socket/uds.html#permissions

An alternative way could be to just try and bind to the socket, but that probably should be done elsewhere?

Reply:
yeah, I think I agree this checking/warning is probably more of a nice to have. Since if they have the socket misconfigured do we prevent successful startup? or do we warn and then default to http url?... which will probably fail to successfully send traces since if they are trying to configure the socket then http will likely fail.


The other/additional approach could be to check the file type, e.g. `file /var/run/datadog/apm.socket` and make sure the resulting type is a `socket` ?

Reply:
Spoke with Kyle, opted to leave it for now and if this seems like an issue, we can add later.